pkg/web3/sdk.go

```diff
@@ -308,3 +312,30 @@ func (sdk *Web3SDK) GetBalance(address string) (*big.Int, error) {
 	}
 	return balance, nil
 }
+
+func (sdk *Web3SDK) GetLPBalance(address string) (*big.Int, error) {
+	// Convert the string address to common.Address
+	client := sdk.Client
+	ethAddress := common.HexToAddress(address)
+	lpToken := common.HexToAddress(sdk.Options.TokenAddress) // LP Token Address
+
+	// Get the balance using the converted address
+	erc20ABIObj, err := abi.JSON(strings.NewReader(erc20ABI))
+	if err != nil {
+		log.Error().Msgf("error parsing ABI: %s", err)
+	}
+	tokenContract := bind.NewBoundContract(lpToken, erc20ABIObj, client, client, client)
+
+	var out []interface{}
+	err = tokenContract.Call(nil, &out, "balanceOf", ethAddress)
+	if err != nil {
+		log.Error().Msgf("error calling balanceOf: %s", err)
+	}
+	lpBalance := *abi.ConvertType(out[0], new(big.Int)).(*big.Int)
```

Comment:
I think the 'out[0]' will throw and out of bounds error here if we enter the err block above it.  Should we return if we're in the error block just above this line?

Reply:
good catch. we should. resolved on the latest commit.